### DISCUSSION TITLE
Inventorising: new demo project

### DISCUSSION THREAD

**Discussion by Aggror** [stride3d/stride#1800] [Contributors]
The current demo project is getting outdated. Meaning we either need to drastically revamp that existing demo project or we need to create something new entirely. For this to happen we need the following:

- What would be a good demo Highlighting Stride's strongest features.
- Who could help out contributing project managemanet, code, art and other topics
- Art style?
- Game play?
- Who wants to contribute? Product owner/manager? Art, Coding, level design, shaders, particles... 

Lets use this topic as a gathering place for new ideas.

**Comment by VaclavElias** [comment]
I can assist only putting docs together, reading, checking, formatting the docs.

**Comment by BlueGrassSquirrel** [comment]
For some reason, I never see much UI handling in the demos.    Stride, flax and some others always want to focus on "look at the rendering capability. Look at how many simultaneous objects we can include!"    And a whole lot of First person views, makes it look like the focus is more on making short films or something vs a game engine. 

a mini RPG Demo could show off both the strong suits, and the fact its a game engine. 

Show off a third person view of a cel shaded player running around a lit forest environment fighting baddies. Complete with HUD, loot/ inventory, particle effects, day night system, interactable grass, leaves wind sway, fire burn, inventory, water, and various animations.     Just one small level/area for the demo.

**Comment by kaikao1208** [comment]
One of thing I will interested will be conducted an animation event through code. For example, character is in idle animation and once CharacterComponent.LinearVelocity.Z is changed character start walking. During walking animation demonstrate how to input footstep SFX in particular frame(second). This will help me on things like more accurate attack event with collider entity enable, VFX during casting magic spell etc..

Another topic I will suggest is  what can await AnimationComponent.Ended(currentAnimation) contribute to wait until animation play to the end. Like I have jump animation(which maybe can also talk in CharacterComponent.isGrounded and CharacterComponent.Jump API) and whenever I landing I will need animation play full landing animation through AnimationComponent.Ended API before into idle.

**Comment by Tonedus** [comment]
So, we want to create a new demo game, that's great. We need to start with general and basic things.

What features and elements of the game engine would we like to demonstrate or focus on: dynamic shadows, particles, animations, physics, PBR, reflections…etc.? 

Also, I would stick to the principle of maximum simplification, for example I would avoid open spaces: forest, streets, nature. If you take other game engines as an example, their demos tend towards closed spaces.

**Comment by Tonedus** [comment]
As an option, I would recommend collecting demos from other game engines as examples. This way we can identify strengths and weaknesses, and understand what direction to take.

**Reply by JermSharWebs** [reply]
Going that route, I'd look at the big two, Unity and Unreal. 
Those two have the important demos, and a few specialties. 

Most of the other engines have demos that are mostly
"Ermahgerd look at Deez grafix!" "Look at how many objects instances we can have!" " realistic lighting? YASSS!" 
The others make you wonder if they're even a game engine at all.  If they do it's usually FPS, side scroll and VR if it can. 

TL; DR:  show off gameplay elements.  
Health, energy bars. Attach one object to another, etc.

**Reply by Tonedus** [reply]
But we don't have the same capabilities as Unity and Unreal engine developers, we are closer to Flax Engine, GODOT, NeoAxis, Torque.

**Reply by JermSharWebs** [reply]
I didn't say to directly copy their demos.
Read the TL;Dr.   Show off gameplay elements. 

A 3D RPG demo.   That would show off floor grids with instanced panels.  Health/energy systems.  UI layouts and usability. 
Read write from files like XML, json.  Save files. 
 Things like that. 
So it would look like a game engine and not an animation renderer or simple level designer.

**Reply by Tonedus** [reply]
I see what you mean. You want the demo to look like a full game with all the gameplay elements, but you need to realize that this is extra effort and time. Although there can be some minimum UI elements if we are talking about an FPS or some action/adventure.

**Reply by JermSharWebs** [reply]
Oh Jesus no not a full game. 
A small block out "level" with cubes for a house, "tree". 
Use capsules for an enemy and an NPC. 
A basic UI using elements available in the engine, to show health and coin. 
Basic character move script that is available in the engine. 
Basic camera movement, with an extra follow player script. 
Basic buy sell script. 
Pick up cube. It stores it in a json. 
Push I for "inventory". Cube is there. 
Sell cube to NPC for 5 coin. Removed that from the json file. 
Buy sphere for 2 coin.  Write to json. Drop sphere on ground. Write that to json. 
"Hit" enemy. It's health go down. It died. 

Ya know. Basic of the basic. 
Things available in the engine, that can make a sample game.  So the Stride looks like a 3d game engine.

**Reply by JermSharWebs** [reply]
Also, if I ever get time off work, I can help with a demo as well.

**Reply by Tonedus** [reply]
In general, I support the idea if there are those who will take on the development of UI and game logic. But here you need to take into account the genre, if it will be an RPG, doing it will be more difficult.

**Comment by Tonedus** [comment]
If nobody minds, I'll post examples of demo games from other engines, maybe not all of them are the best example, but for clarity it will be useful. Demos are both official and fan-made.

**Comment by Tonedus** [comment]
https://www.youtube.com/watch?v=-LR6Rjx0hAI
https://www.youtube.com/watch?v=KK8oL9_5hVo&t=163s 
https://www.youtube.com/watch?v=22zplE1STgU
https://www.youtube.com/watch?v=_KCl_m1IGp4&t=3s
https://www.youtube.com/watch?v=xRHsTbB24VM
https://www.youtube.com/watch?v=2ShnOuSFJsA
https://www.youtube.com/watch?v=PBktSo0bXas
https://www.youtube.com/watch?v=3sPFBmnfEEM
https://www.youtube.com/watch?v=Fm9a6FGBWbs
https://www.youtube.com/watch?v=Nn0coOoh35E
https://www.youtube.com/watch?v=Ngbwn_T6hnE
https://www.youtube.com/watch?v=tpKCo4Jcarc&t=332s

**Reply by JermSharWebs** [reply]
Renderer
Renderer uggh...your not Blender, stahp.
Renderer and a few features. 
That flax FPs demo was good. 
 UI, particles, gameplay logic and rendering all in one demo.    

Unreal has the budget to go wild. Mostly a rendering demo for them lol.
But yeah, UI, game play logic, particles, animation.

**Comment by Tonedus** [comment]
More specifically - it seems to me that the best option would be action/adventures in isometric view with closed locations.

Isometric view will allow to neglect high texture detail that will slightly simplify the task, the ability to see the character will allow to demonstrate his animation and will be a plus when you can see how the character interacts with the environment. Closed locations will not scatter attention and better work out the interior.

**Comment by VaclavElias** [comment]
I would also suggest maybe multiple demos for different coding and complexity level? With each level, adding more features and complexity? It could be just one game, going through these levels, or multiple games, or mix.

Maybe coding complexity level 1-10?

**Comment by fydar** [comment]
For a demo that's an outside space but not an open world; Unity put together this demo scene that reminds me an awful lot of a Destiny map.

<p align="center">
<a href="https://youtu.be/ARorKHRTI80">
<img width="512px" src="https://github.com/stride3d/stride/assets/19309165/b70e50aa-2f8a-403a-9a4e-31faff41ee4a" />
</a>
<br/>
<sub>
<a href="https://youtu.be/ARorKHRTI80">https://youtu.be/ARorKHRTI80</a>
</sub>
</p>

**Reply by Tonedus** [reply]
I like this (Blade Runner) visual style of architecture combining something old and at the same moment modern sci-fi elements, it's worth considering this option.

**Reply by fydar** [reply]
The style of architecture is called **Brutalist Architecture** and is explicitly what Blade Runner 2049 was referencing (specifically Soviet-era Brutalist Architecture).

<p align="center">
<img width="400px" src="https://github.com/stride3d/stride/assets/19309165/bfa9bf2f-959c-4073-ba45-5be26c155eab" />
</p>

<p align="center">
<img width="400px" src="https://github.com/stride3d/stride/assets/19309165/eae9f9fb-14cf-4fc4-9809-a753b4b2dc3a" />
</p>

<p align="center">
<img width="400px" src="https://github.com/stride3d/stride/assets/19309165/dbb8b174-be16-4a11-93d1-0532f613762d" />
</p>

**Reply by Tonedus** [reply]
This style is definitely worth taking into consideration, for something sci-fi would be great. What do you think about the idea for the demo? What about the genre? What would it be like?

**Reply by Tonedus** [reply]
Dune also has a similar architectural style.
![d0468ba063e695f00755d25cf2b5fe78.jpg](https://github.com/stride3d/stride/assets/42056644/69277250-664a-41f7-a41b-02b03850d1d2)

![DU-29026.jpg](https://github.com/stride3d/stride/assets/42056644/61f4e390-b10b-433a-8fc8-1d2b0c0a1321)

![dune-set-design-29167-1635370279.jpg](https://github.com/stride3d/stride/assets/42056644/8eed8af5-6c11-4803-9eb8-f48a45232eee)

![8bc28affc88d08214d41d11fef94c505.jpg](https://github.com/stride3d/stride/assets/42056644/013c1c56-24f4-4984-b0f2-2ac56671dc13)

![0732a2b51d71f05ff29472248b31948084e62800-3000x1128.jpg](https://github.com/stride3d/stride/assets/42056644/14ab98ab-c01f-422e-8bfe-ea1b6dd4a960)

**Comment by Tonedus** [comment]
Then maybe we can borrow something from the brutalist architectural style and bring something from sci-fi, it'll be a vibe, with references to Dune/Blade Runner. It could be one building that would be interesting to explore.

**Reply by Aggror** [reply]
Would you be willing to lead this project? Doesn't even have to be code or art, but someone who takes charge of getting everything together and talks to people wanting to contribute.

**Reply by Tonedus** [reply]
Although I have no experience in the position of a lead, I could try to take on this issue. All I can do on my part is to hold a number of discussions with all those interested in it and systematize all the information about the demo. And as a 3D modeler I can be directly involved in creating 3D content for the demo.

**Reply by Aggror** [reply]
That is fine, we are all just doing our best to make something really cool of it all. 
Having someone who follows up on people, who said they wanted to contribute in any kind of way to a demo, is important. It keeps momentum going.
And you can always message me for questions/advice of course

**Comment by BlueGrassSquirrel** [comment]
So is this going to be an all in one show off as much as possible demo? 
tbh I'm still getting the hang of stride's namings and way of doing things. 
like getting a screen to world ray is a whole thing. modifying the "rpg demo" example to output the ray hit as it hit, vs mouse click to track the mouse for dragging item around the in game world. 

OOOOHHHHhhh hehe hehe so I've been keeping a personal log of "unity scripts converted to stride".    I should post that one somewhere and update it as I go.   It's the built in Unity functions names vs what its called in stride(if it exists).  With some notes on what features has to be custom scripted.  Its very slow going though as I don't get much time off work to do anything at all.

**Comment by carl2049** [comment]
Let's go with a space battle near a black hole?

**Reply by Aggror** [reply]
Hi @carl2049 
Can you elaborate more? Maybe draw some rough shetches, generate some art or use existing images as a reference? Any gameplay? Just a cut scene?

**Reply by carl2049** [reply]
For now it's just an idea in my head. This is something similar to the Expanse last season that takes place at the Ring, but instead of Ring it's a black hole. The gameplay can be a potential showcase for first/third person shooter, space nebula environment, vehicle mechanics, particles effects. etc. And my favorite part, physics engine showcase where the violent black hole keeps sucking everything in while you have stay afloat controlling the ship/pod alongside battling the enemy.  Sigh!

**Reply by Idomeneas1970** [reply]
Hi @Tonedus,
Idea for small 3d demo:
1) Entry cut scene setting the mood, showcasing atmosphere/lighting effects, music and even voice over.
2) Menu screen with different options: start game, load/save game, multiplayer, credits.
3) Game play: 
a) illustrate hud options (health, stamina bars, ammo counts, attack/armor ratings etc etc), camera options (first, 3-d person, over shoulder, isometric etc). swap on pressing, say, k-key, zoom in/out etc.
b) illustrate animations: walk, run, attacking, getting hurt, jumping, falling, dying, etc etc
c) illustrate world interactions, from dialogs with npcs (voice overs included), attacking a foe, inventory while opening a chest in the world etc
d) illustrate multiplayer capabilities (server-client connections), at least two players interacting with the world, or talking to each other (shaded area above their heads with dialog).
e) setting can be indoors for all this, but it would be great to include a massive outdoors camera panning like the world from the TR-stride sample. That outdoors is impressive. Add trees, foliage etc.

A lot of these concepts exist in stride/xenko samples all over the place; they need to be put together in a cohesive project.
I will be glad to help best I can @Tonedus , but I am still learning the stride basics. We can talk about the plot/specifics on discord. All I can give is time (when I have it), so we can get this going asap.
Cheers

**Comment by IXLLEGACYIXL** [comment]
as we cant hire an artist from the open collective we can use other free tools 

for materials we can use https://www.blenderkit.com/
It has many free materials and also some halfway usable meshs

for better meshs we can use https://www.sloyd.ai/
you can generate good meshes in seconds, its not really an AI
a sloyd is for example an axe and you configure what you want and then press generate and it will make meshes generate, 
a sloyd is free in usage , you just cant sell a generated mesh on its own but we dont make money with the demo anyway so that wont be a problem at all
there we can generate meshes in seconds and they release weekly new sloyds 
you pick something
![grafik](https://github.com/stride3d/stride/assets/107197024/d25aac1d-9d1f-4393-b9dc-2848d82b8a3e)
![grafik](https://github.com/stride3d/stride/assets/107197024/b13b4f0f-e071-43c1-a63d-eb90aa952bb0)
this is for example a genereated battle ram , now i click on randomizer and generate again
![grafik](https://github.com/stride3d/stride/assets/107197024/859a1b0a-3925-4fc6-89d2-5ef914c1de63)

its primary focus is on items and not human models

they have also some materials

**Reply by Tonedus** [reply]
Thank you for these resources, I think we will definitely use some ready-made solutions to fill the scene with 3D content.

**Comment by Tonedus** [comment]
In addition to everything else, we have a number of organisational issues. Some people continue to insist on a separate channel in Discord, arguing that demo is a more specific and concrete topic and it requires a separate chat. [Agrror](https://github.com/Aggror) has repeatedly commented on this issue, the showcasing channel can be fully dedicated to this topic. If there are other suggestions, please let me know.

The second issue is the specific direction that participants can take, we need those who are willing to take on code, animation, level designer, etc, here you can note what role you are willing to take on, I will contact everyone to find out the details.

**Reply by Aggror** [reply]
I can rename the  'showcasing' channel if you want. Demo scene 2024 for instance

**Comment by Keepsie** [comment]
I think the best idea is what @JermSharWebs said which is a small RPG demo. This is what got me started in unity years ago.
https://youtu.be/0Sh6WncKUmI?si=kOpYyyQbfhlXaybi

This is simple and showcases many use cases for new users. It can also be used as an example to showcase Stride in marketing.

It just needs to be a basic level with basic functionality. We are pretty busy, but I have a few 3D artists on my team. I'm sure I can have one of them or myself create a modular kit if the theme isn't something that's going to take forever to make. Or just help with some coding if you guys have list of mechanics that's needed.

**Reply by Tonedus** [reply]
Great, 3D artists will be very useful for us! I don't think the scene for the demo will be large, we'll stick to a small space, which will allow us to focus more effort on it.

**Comment by WithBlueHorns** [comment]
Hey, I'm available for some work, I can do the art aspect of the demo

**Reply by Tonedus** [reply]
Thanks, great! I believe we'll probably need some preliminary art for the demo scene design.

**Reply by WithBlueHorns** [reply]
Hey, if possible, if y'all will make your minds and will need assets (assuming you are in charge of it) please contact me on discord. I rarely look at github so that might speed up it, a lot

**Reply by Tonedus** [reply]
Got it!

**Comment by ArieLeo** [comment]
Hmm how about an interactive showcase/showroom building for a tech demo/ demo scene, similar to a museum or art gallery. To showcase the tech and feature of stride.
- A building that have multiple room and floor in brutalism architectural style, to show case scene streaming and lighting.
- A simple interaction system to interact with the showcase, ie basic interaction press e/f to interact with a button entity and trigger an event/function script on another entity / self.

This way in the future when Stride have a new feature to showcase it can be added to the showroom or by creating a new room/ floor and expand the demo scene.

**Reply by Tonedus** [reply]
I think I saw something like this about the Flax engine, a very good example!

**Reply by ArieLeo** [reply]
Yeah flax have this as a project sample, but what i have in mind is something like UE showroom project.
If i'm not mistaken UE showroom project are quite large (in scale not project file size) and have multiple rooms, and everytime there's a new feature it always added into that project.

**Comment by Tonedus** [comment]
So, some time has passed and I think we can summarize some intermediate results. Some participants have joined us, they have offered interesting ideas and suggestions, thanks for that, some of them are ready to help with writing code, some of them are ready to help with 3D content and art, I will contact some of them again later and clarify what kind of help they are ready to offer. 

And now the conclusion about ideas and suggestions for the demo. All suggestions can be categorized into two types:

- Demonstration of game elements (UI, HUD, loot, inventory...)
- Interactive showcase/demo room

We should take into account that we have to proceed from our capabilities, I would suggest to stick to the principle - from simple to complex. It would be correct at the first stage to choose the simplest possible basic idea from which we can proceed in the future, such an idea can be a small space or building, the purpose of the demo will be to explore certain places that demonstrate the features of the engine, moving through such a space can be carried out with the help of a first/third-person controller or camera. For example, we can make a building consisting of several rooms, each of which will be a separate demonstration of technology.

**Reply by VaclavElias** [reply]
From simple to complex is very good idea, because we can ship sooner these demos, while gradual complexity is increased 🙂

**Reply by Jarb2104** [reply]
That gives me the idea to make a hotel with an elevator that works similar to how the elevator of the inception movie works, this way we can create several small demos, each being a single level in the elevator, which can also make it easier to add or remove levels from the elevator later on. This can also help create different types of demos, levels 1 through 3 were the sci-fi buildings, but suddenly level 4 it's an action adventure platforming or something. 

And we even can ship or showcase the demo earlier with not all levels completed, following the idea it could be that each level also has more complexity as you go up.

**Reply by Tonedus** [reply]
The elevator idea is great, but do we have the effort to create several variations of the demo at once?  Maybe each level will have a different technology/engine feature? For example: 1- physics  2 - particles 3 - dynamic shadows/lighting, etc.

**Reply by Jarb2104** [reply]
Yeah, that's the idea, and like I said this opens the demo to the possibility of shipping it with only basics, for example, and keep adding things later on, it can even be adjusted in the future as the engine evolves.

And you can even pick better scenes to showcase the different technologies or features of the engine. Maybe a forest for the lightning shadows, but use a house on fire for the particles.

Always with the possibility to do something "gamie" in each scene, like simply run around the object picking stuff, or putting out fires in the other.

This doesn't have to increase the scope, as it should focus on the specific thing, and again we can "release" it with only 2 or 3 levels with the very basics, and later on add or remove levels as needed.

P.S. I am open to help with the efforts coding, I don't know much about the engine yet, or it's best practices, but it'll help me learn those as well.

**Comment by JoeAbiN** [comment]
If you're going for a sci-fi environment, I suggest using drones as simple enemies or maybe just background elements. They can have a simple AI behaviour or boid simulation. They can add an arcade-y feel to the game with simple yet engaging gameplay (for example you shoot them down). I've been trying to do a simple boid simulation but didn't have much time to develop it. I don't know if that idea is worth exploring.

**Comment by Tonedus** [comment]
At the moment, summarizing all the suggestions, we have two main options from which we should proceed. 
1. Demonstration of game elements 
2. Interactive showcase/demo room

We should decide on the option and move on.

**Comment by Tonedus** [comment]
Simplified game

- In this case we make some kind of building, room or space where controlling the character can interact with the environment, in this case can be realized some elements of the game. Some prototype of a first/third-person game in a simplified form.

Interactive showcase/demonstration room

- In this case we make some kind of room or space that can be explored with a camera or a scripted camera and watch the demonstration of various features of the engine.

**Reply by Jarb2104** [reply]
I am inclined to use the first approach, since Stride is a game engine after all, a hands on demo where you can control a character and interact with stuff would be the best, IMHO, plus making it available for people to toy with it can draw a lot of new users.

**Comment by CodeLikeCXK** [comment]
What about a stylized arpg with full enemy AI behavior tree system?
A good chance of showing as well as improving AI and animation

**Reply by CodeLikeCXK** [reply]
Hack-and-slash through smart enemies in brutalist world

**Reply by Tonedus** [reply]
Are you talking about a small game with all the gameplay elements? We have to stick to the principle of simple to complex. Right now, I don't see any way for us to realize something like that.

**Reply by CodeLikeCXK** [reply]
OK.Sorry.

**Comment by Tonedus** [comment]
Taking into account all suggestions and discussions, I will try to form a vision of how our new demo could look like. So, we came to a consensus that the general idea should be a simplified game, we will start from that. Since we adhere to the principle of simplification, the best choice would be a closed space or a building, the creation of the external environment will take much more time and effort. 
Now about the gameplay for the demo. I don't have a clear opinion on it yet, but I can offer a general concept. The basis can be a third-person shooter in a sci-fi setting. The basis of gameplay can be the passage of levels in the form of floors of a building, to move between floors can be realized elevator, it will allow each floor to make modular, it will allow something to add or change in the future. Enemies as in the old demo can be represented as robots/drones, but on each floor the difficulty level of the enemy increases, access to each next floor is blocked until the player destroys all the drones, on each floor can be a key card with access to the next floor. Each floor can be a demonstration of a new technology, effect or engine feature, among other things. The last top floor can be represented by the final boss drone.

**Reply by VaclavElias** [reply]
I like robots/drones idea :)

**Comment by Tonedus** [comment]
With the help of the AI I have created a small set of images that can be used as inspiration when creating levels for the demo. A lot of people have suggested using Brutalist architecture with its monumental forms, I think this is a good idea that we can take on board, it would be nice to get a Dune/Blade Runner style vibe.

I'd like to ask all participants to give their opinion on the concept for the demo, maybe someone wants to add to it or suggest something of their own.
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_0 (4)](https://github.com/stride3d/stride/assets/42056644/5cf1ec06-1e23-4297-91ac-c96a0c02b5c2)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_0 (5)](https://github.com/stride3d/stride/assets/42056644/3697e6e2-5770-42f0-b033-87bd00b53c06)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_0 (6)](https://github.com/stride3d/stride/assets/42056644/12e9e56d-1507-4fa3-8ec5-007efd8d45af)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_0 (9)](https://github.com/stride3d/stride/assets/42056644/a056322c-4427-4b9d-a3bd-21a152120bcf)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_0 (10)](https://github.com/stride3d/stride/assets/42056644/274d3a9c-8dae-49a2-ba44-891e169ffab5)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_1 (3)](https://github.com/stride3d/stride/assets/42056644/30225629-2f86-4c28-89b8-4c69e1cde16d)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_1 (4)](https://github.com/stride3d/stride/assets/42056644/63023cd9-0873-460b-9347-225962d25395)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_1 (6)](https://github.com/stride3d/stride/assets/42056644/7e047166-8b71-497e-9543-e17dcb76c511)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_2 (3)](https://github.com/stride3d/stride/assets/42056644/48cd14fa-4821-4606-bb4f-e2323a2bbd93)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_2 (4)](https://github.com/stride3d/stride/assets/42056644/e7e9222d-2ec9-49bd-8dd2-9fe5fc8b06fd)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_3 (2)](https://github.com/stride3d/stride/assets/42056644/2be75989-bf25-4bd8-90cb-d7203529ef46)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_3 (3)](https://github.com/stride3d/stride/assets/42056644/8390ea3f-68f3-45a1-a041-aec715f66d9c)
![Leonardo_Vision_XL_The_event_takes_place_in_the_future_A_large_3 (4)](https://github.com/stride3d/stride/assets/42056644/b65a6f34-bf7d-439e-975d-3886b44d90e5)

**Reply by VaclavElias** [reply]
I like it. I would add a spark of wear and tear, showing an age of such monumental architecture.

**Comment by Tonedus** [comment]
So, there was time to think more about the details of the concept. The location design itself is still in the process of final approval, but as for the general concept - I propose to leave unchanged the idea of rooms that need to be passed in order of difficulty. As enemies I propose to stop on drones/turrets. As for the difference between the rooms, for maximum variety we can realize:
- different difficulty level of enemies
- different weapons for the character
- adding a system of cover from enemy fire, stealth mode
- droid hacking
- utilization of physical capabilities of objects (throwing objects with the help of gravi-weapons, shooting at explosive containers)

And now I'll try to describe how it roughly can look like:
The character appears in a hall with several rooms, all rooms are closed except for one, this is signaled by a light bulb above the door. In the hall there are several crates with weapons, but the player initially has access to one weapon, the other crates are closed. After successfully passing the first room, the second door in the order opens, as well as the second weapon box.   

Below I have modeled a general blockout of the location for the demo, one of the possible variants.
![2023-12-13_191431](https://github.com/stride3d/stride/assets/42056644/28d1d558-809a-402b-9075-3a8142bcee9a)
![2023-12-13_191711](https://github.com/stride3d/stride/assets/42056644/708b6f2f-dcf7-4f12-9401-2fc4c8875906)

**Reply by Doprez** [reply]
What would the type of player movement need to be? is it going to be fine to have a simple charcter controller that jumps and moves or do we want a Quake like controller for air strafing and such?

This looks great so far!

**Reply by Tonedus** [reply]
I think the demo needs a standard third-person controller that will allow you to run, jump, and hide behind cover in addition to using weapons.

**Reply by Doprez** [reply]
I like it, it will allow a good showing of animations with third person.

**Reply by Aggror** [reply]
I like your room layout and the unlocking element. 

As for gameplay: try and keep it simple. It seems all too easy on paper, but getting something out door that feels nice to play costs effort. Personally I the list of features is already on the large side. Personally I would already scratch the last 2/3 items for the time being.
